docs/references/architecture/adr-119-dog-mempool-gossip.md

```diff
@@ -0,0 +1,367 @@
+`HaveTx` messages if the redundancy is too high. 
+
+Once a `HaveTx` message is sent,
+sending further `HaveTx` messages is blocked. It is unblocked by the redundancy control mechanism 
+once it is triggered again. This will be discussed below, but overall the frequency
+of `HaveTx` messages should be aligned with the time it takes to send a message and 
+adjust routing between peers. 
+
+`broadcastTxRoutine`
+
+Before sending a transaction, we filter the peers to send to based on the `disabledRoutes`. Again,
+only if DOG is enabled. 
+
+`RemovePeer`
+
+When a peer is removed, we remove any existing entries related to this peer from the `disabledRoutes` 
+map and explicitly trigger redundancy re-adjustment. The later is not strictly needed, but 
+can lead to quicker propagation of this information through the network.
+
+Note that we had considered a scenario where, upon removing a peer, a node sends `ResetRoute` messages to all its peers. 
+This lead to routes being disabled too many times, and the process of stabilizing redundancy would be re-triggered
+too frequently. The adapted logic relies in the redundancy controller to decide whether a `ResetRoute` 
+message should be sent. 
+
+
+#### Monitoring
+
+The impact of the protocol on operations can also be observed by looking at the following metrics that already exist:
+
+- `mempool.already_received_txs` - the number of redundant transactions. When DOG is enabled these values should drop. 
+
+- `p2p.message_send_bytes_total` - This metric shows the number of bytes sent per message type. Without DOG, the transactions tend to dominate the number of bytes, while, when enabled, the block parts should dominate it.
+
+- `p2p.message_receive_bytes_total` - This metric shows the number of bytes received per message type. As with the metric abobe, with DOG the dominating messages should be block parts. 
+
+The received and sent bytes metrics are reported by message type which enables operators to observe the number of `HaveTx` and `ResetRoute` messages sent.
+
+
+This ADR introduces a set of metrics into the `mempool` module. They can be used to observe the parameters of the protocol: 
+
+
+- `mempool.disabled_routes` - Number of disabled routes.
+
+- `mempool.redundancy` -  The current level of redundancy computed as the ratio between duplicate and unique transactions. 
+
+
+### Configuration parameters
+
+
+- `mempool.dog_protocol_enabled: bool`: Enabling or disabling the DOG protocol. `true` by default.
+The only reason for this is the incompatibility of DOG with the existing experimental feature
+that disables sending transactions to all peers. 
+
+
+The [specification](https://github.com/cometbft/cometbft/spec/mempool/gossip/dog.md) of the protocol introduces 3 additional variables. 
+Out of the tree, we have made the following configuration parameters:
+ 
+- `mempool.dog_adjust_interval: time.Duration`: Set to `1s` by default. Indicates how often the redundancy controller readjusts 
```

Comment:
What does it depend on? Does it have to be a configuration parameter?

Reply:
We ran experiments varying number of peers, tx load and target redundancy and did not see a strong indicator to change this to a particularly lower or higher value. But we left it as a config because it might be that for some users this can be too frequent.